### DISCUSSION TITLE
Nightscout not alarming correctly

### DISCUSSION THREAD

**Discussion by gregbeach** [NightscoutFoundation/xDrip#3807] [Q&A]
Hi - not sure if this is the right place to ask this question - tried the CGM facebook group and haven't had success.  I recently upgraded my Google Cloud deployment of Nightscout to 15.0.1.  I think I originally deployed Nightscout on Google Cloud in Dec 2022.   I've always had the alarms off but now I want to get them to work and it's not going well.  The urgent low and high alarms go off when my son's BG isn't even close to a high or low (or predicted to be high or low by loop).  They won't go off when he is actually urgently high or low.  I'd really appreciate it if anyone could help me fix this or let me know if this is a known problem.  My son is running loop.  He has a Dexcom G7.  My NS variables are below.  Thanks!

BTW - the main point in enabling the alarms is because my teenage son doesn't wake-up for low alarms from the Dexcom app on his phone and he'll be going to college soon (very scary). He doesn't really wake up to any alarm at this point.  I was planning on buying the sonic bomb alarm clock, attaching it to a smart plug, and using IFTT using the Nightscout urgent low alarm to make it go off and vibrate.  Before I do this, I need to know if the NS urgent low alarm is reliable.

export ENABLE='careportal food boluscalc bwp cob bgi pump openaps rawbg iob upbat cage sage basal loop override>
export AUTH_DEFAULT_ROLES='denied'
export SHOW_FORECAST='loop'
export ALARM_TYPES='simple'
export BG_HIGH='300'
export BG_LOW='55'
export BG_TARGET_BOTTOM='80'
export BG_TARGET_TOP='250'
export DISPLAY_UNITS='mg/dl'
export ALARM_HIGH='off'
export ALARM_LOW='off'
export ALARM_TIMEAGO_URGENT='off'
export ALARM_TIMEAGO_WARN='off'
export ALARM_URGENT_HIGH='off'
export ALARM_URGENT_LOW='off'
export BOLUS_RENDER_OVER='0.1'
export THEME='colors'
export SHOW_PLUGINS='loop iob cob pump loop override basal dbsize'
export CUSTOM_TITLE='DIABETES_SUCKS'
export PUMP_FIELDS='reservoir battery clock'
export DEVICESTATUS_ADVANCED='true'
export ALARM_TYPE_PREDICT='0'
export DBSIZE_MAX='20000'
export BASAL_RENDER='default'

**Comment by Navid200** [comment]
When you open Nightscout in a browser, do you not see the in range values shown with green dots and high or low values in yellow and urgent high and urgent low values in red?

**Reply by gregbeach** [reply]
Yes, the graph all looks right.  Green dots when in range with the yellow and red dots as expected according to my values.  The settings dropdown menu has the right values in it and I have the relevant alarms checked (as my config has these variables off by default). Still, the alarms don't work when they should be alarming and do alarm at times when they shouldn't - it seems random.  Thanks for any help you can provide.

**Reply by gregbeach** [reply]
Also, the nightscout app on my iphone has the same issues - so whatever is happening isn't browser-specific.

**Reply by Navid200** [reply]
In my perspective, Nightscout is a server, not an app.  I wouldn't expect a server to alert me.
If the only way Nightscout would alert me is when I have it open in a browser, how will I get an alert if the browser is closed?

I get my alerts from xDrip, an app.
xDrip has many different alert features that may be useful.  For example, you can set it up to flash:  https://navid200.github.io/xDrip/docs/Alerts/TurnLightOn.html

**Reply by gregbeach** [reply]
My son uses Loop on his iphone.  We've only been using Dexcom alerts for the last 10+ years.  However, he'll be off to college next year and the Dexcom alarms have proven incapable of waking him up at night.  If his number is low, I get the alert and then go wake him up and give him juice.  Starting next year and hopefully before this, we need something that has a better chance of waking him up if he goes low.  Nightscout supports IFTT.  My thinking is to get the sonic bomb alarm clock which also has a vibrating pad.  Connect this to a smart plug (e.g. Wyze plug) and when Nightscout gets an urgent low number, then use IFTTT to power on the plug to get the alarm/vibration.  I've seen the sugar pixel but I highly doubt he wants to see a big display of his number (he hates being reminded that he has diabetes).  In any case, my solution would require a reliable urgent low signal from nightscout.  Make sense?

**Reply by Navid200** [reply]
What happens if he turns off the sonic bomb alarm clock and forgets to turn it on?  
What happens if he is staying at someone else's dorm after a study get-together while the clock is at his dorm?
Wouldn't it be better to have everything on his phone?

You can set up xDrip to vibrate.  You can set up xDrip to generate obnoxious alert sounds if you want to.  

If using Dexcom app is a necessity, you can still use xDrip in companion app mode.

**Reply by gregbeach** [reply]
I'm not sure.....I suppose I can setup IFTTT to turn on and then off after a set time of triggering it - and then it will trigger again if number is still high.  Yes, not ideal.  He would be taking a risk if he slept somewhere else.  I don't think a phone vibrating will be enough, if that's what you mean.  He's also going to be reluctant to have his alarm on full blast if there is roommate involved.  I don't know much about xDrip, I'll look into it.  I think the ideal situation is having his bed shake or something violently vibrating right next to him.  He's not going to sleep with his phone in his pocket.  Can Xdrip trigger something else to vibrate?

**Reply by Navid200** [reply]
I don't think xDrip can trigger something else to vibrate.

xDrip has ascending volume profile:  https://navid200.github.io/xDrip/docs/Ascending-volume-profile.html
It plays the alert starting t low volume.  Once a minute it plays the same alert again and every time it increases the volume.  You can set it up for this to increase the volume all the way to max.  

With respect to your original question, I have to say that it is related directly with Nightscout and not xDrip.
This group set up Nightscout on Google Cloud.  But, Nightscout has its own support groups.  They must be able to answer your questions better than we can here.

I'm sorry if you have already answered this.  What about wearing a watch when he sleeps and the watch vibrates on his wrist for an alert.
There is a BlueJay watch that is till going through beta tests.  It runs xDrip.  It can get readings directly from a G7.  I cannot tell you that it can coexist with Dexcom app.  You have to ask that question from the company.  Search for BlueJay Watch.  The model you want to consider is U2.  

I have diabetes and know that one size does not fit all and would like to thank you for taking the time to help someone who has diabetes and also for providing feedback to us.
We continue to improve xDrip when possible.

I will let you know if I can think of anything else.

**Reply by gregbeach** [reply]
Thank you for answering but also for all the heavy lifting with Nightscout on Google Cloud (and other stuff you are doing with Xdrip that I'm not leveraging).  The bottom line for our son is he's a very heavy sleeper and I know he'll be conscious of waking up any roommates with loud audible alarms.  The iphone can get very loud for his normal wake-up alarms and that still usually doesn't work.  He has an apple watch he could use for vibration but I doubt it's enough.  For my nightscout alarms, I can only guess that something is screwed up with the way loop is uploading his BG number to nightscout (loop uploads all the info instead of Nightscout getting the number from the Dexcom server).  I forget why this is preferable but I think there's a config in loop to get the number just from Dexcom.  I see there are two solutions in the market.  1) iLuv Smart Shaker.  This is a vibrating disc that uses bluetooth and can be setup to vibrate on dexcom alerts.  Not sure this will vibrate enough though, and 2) Sugar Pixel.  This has a vibrating disc.  Both may work so I might go that route.
